### DISCUSSION TITLE
Are the paid plans faster?

### DISCUSSION THREAD

**Discussion by ejkreboot** [vercel/community#1797] [Help] [answered]
### Summary

I am experiencing somewhat slow performance on my sveltekit app deployed to vercel (free plan). By this I mean that delivering a <5kb file from my supabase storage bucket via my api hosted on vercel is taking 2-4 seconds (retrieving the same resource via the app running on my local machine is nearly instantaneous). 

I am curious if the paid tiers have more network and compute resources dedicated to them? If I upgrade might the performance upgrade as well?

(For anyone interested I will paste the code for this endpoint below.)

### Additional information

```
import { supabaseClient as supabase } from '$lib/supabase.js';

export async function GET({ params }) {
    const path = params.post + "/post"
    const result = await supabase
        .storage
        .from('blog-posts')
        .download(path)
    const blob = result.data;
    const buffer = Buffer.from( await blob.arrayBuffer() );
    return new Response(buffer);
}
```

**Comment by leerob** [comment]
A Vercel Function, [automatically generated](https://vercel.com/blog/framework-defined-infrastructure) from your SvelteKit route (the `GET`), uses the same infrastructure on the free and paid plans. There are two different runtimes: Node.js (Serverless Functions) and Edge (Edge Functions). You can use both with SvelteKit on Vercel applications.

Could you confirm that your Vercel Function is in the same region as your Supabase database? The default for Serverless Functions is US East. I believe this is also the default in Supabase, but you might have changed it. This could explain why it's faster on your machine versus needing to have further network hops from the USA -> Europe, for example, if incorrectly configured.

**Comment by ejkreboot** [comment, marked as answer]
Oh...good point about the serverless function region. My supabase project is hosted on us-west-1. I changed the function region in the functions settings of my vercel project to US West (sfo1), and indeed the response times are much snappier now. Thank you!

**Reply by leerob** [reply]
Woohoo! Glad to hear it.
